splice_machine/src/main/java/com/splicemachine/derby/impl/storage/SpliceRegionAdmin.java

```diff
@@ -81,6 +82,78 @@
 
     private static final Logger LOG=Logger.getLogger(SpliceRegionAdmin.class);
 
+
+    public static void DELETE_REGION(String schemaName,
+                                     String tableName,
+                                     String indexName,
+                                     String encodedRegionName) throws Exception {
+        TableDescriptor td = getTableDescriptor(schemaName, tableName);
+        ConglomerateDescriptor index = null;
+        if (indexName != null) {
+            indexName = indexName.trim();
+            index = getIndex(td, indexName);
+            if (index == null) {
+                throw StandardException.newException(SQLState.LANG_INDEX_NOT_FOUND, indexName);
+            }
+        }
+
+        if (encodedRegionName == null) {
+            throw StandardException.newException(SQLState.PARAMETER_CANNOT_BE_NULL, "encodedRegionName");
+        }
+        String conglomId = Long.toString(index == null ? td.getHeapConglomerateId() : index.getConglomerateNumber());
+        Configuration conf = (Configuration) SIDriver.driver().getConfiguration().getConfigSource().unwrapDelegate();
+        PartitionAdmin admin = SIDriver.driver().getTableFactory().getAdmin();
+        String partitionName = null;
+
+        boolean regionClosed = false;
+        boolean fileMoved = false;
+        Partition p = getPartition(td, index, encodedRegionName);
+        try {
+            // close the region
+            partitionName = p.getName();
+            regionClosed = true;
+            admin.closeRegion(p);
+            SpliceLogUtils.info(LOG, "Closed region %s", partitionName);
+
+            // move all store files to a temporary directory
+            fileMoved = true;
```

Comment:
should "fileMoved" to be set after the call of "moveToArchive()"?

Reply:
Before it to be safe, just in case it fails when moving files. Then clean up by moving files back.